examples/DallasTemperatureSensor/DallasTemperatureSensor.ino

```diff
@@ -22,88 +22,179 @@
-     present(i, S_TEMP);
-  }
+void loop()
+{
+
+   // You should not change these variables:
+   static boolean isMeasuring = true;                        // Used to indicate when the time is right for a new measurement to be made.
+   static boolean isCalculating = false;                     // Used to bridge the time that is needed to calculate the temperature values by the Dallas library.
+   static unsigned long currentMillis = 0;                   // The millisecond clock in the main loop.
+   static unsigned long previousMeasurementMillis = 0;       // Used to remember the time of the last temperature measurement.
+   static int16_t conversionTime = 0;                        // Used to store the time needed to calculate the temperature from measurements.
+
+   currentMillis = millis(); // The time since the sensor started, counted in milliseconds. This script tries to avoid using the Sleep function, so that it could at the same time be a MySensors repeater.
+
+   // Let's measure the temperature
+   if(isMeasuring == true && currentMillis - previousMeasurementMillis >= measurementInterval) // If we're not calculating, and enough time has passed, we'll start again.
+   { 
+      isMeasuring = false; // We're measuring, so let's take it off our to-do list.
+      Serial.print("Starting new measurement(s)\n");
+      previousMeasurementMillis = currentMillis; // Mark the time of the initialiation of this measurement.
+
+      // Fetch temperatures from Dallas sensors
+      sensors.requestTemperatures();
+
+      // query conversion time. Apparently it takes a while to calculate.
+      //ConversionTime = sensors.millisToWaitForConversion(sensors.getResolution());
+      conversionTime = millisToWaitForConversion(sensors.getResolution()); // This is a modified version of the line above, to deal with the problem in the current Dallas library.
+      isCalculating = true; //Next step is to re-calculate the temperature again.
+   }
+
+
+   // Next, let's calculate and send the temperature
+   if(isCalculating == true && currentMillis > previousMeasurementMillis + conversionTime ) 
+   {
+      isCalculating = false; // We're doing this now, so check calculating off the to-do list too.
+      for (int i=0; i<numSensors && i<maxAttachedDS18B20; i++)  // Loop through all the attached temperature sensors.
+      {   
+         
+#ifdef FAHRENHEIT // Arduino's pre-processor selects which code gets added to the sensor based on which define is set in the settings at the top of this sketch.
+         float temperature = sensors.getTempFByIndex(i);
+#else
+         float temperature = sensors.getTempCByIndex(i);
+#endif
+         
+         Serial.print("Sensor #");
+         Serial.print(i);
+         Serial.print(" says it is ");
+         Serial.print(temperature);
+         Serial.print(" degrees\n");
+         if(temperature != -127.00 && temperature != 85.00) // Avoids working with measurement errors.
```

Comment:
Maybe we should use:  
DEVICE_DISCONNECTED_C
or 
DEVICE_DISCONNECTED_F 
defined in DallasTemperature.h

which one depends on if getTempCByIndex / getTempFByIndex was called.